source/transactions/tests/README.rst

```diff
@@ -97,7 +105,9 @@ Each YAML file has the following keys:
   - ``operations``: Array of documents, each describing an operation to be
     executed. Each document has the following fields:
 
-    - ``name``: The name of the operation on ``object``.
+    - ``name``: The name of the operation on ``object`` or the name of a
+      special test operation to run, eg. "failPoint". See
+      `Special Test Operations`_.
```

Comment:
We decided not to YAML-ify the live failover spec tests, so I don't think this is necessary.

Reply:
I already changed the format from:
```
      - name: failPoint
        session: session0
        failPoint:
          configureFailPoint: failCommand
          mode: { times: 1 }
          data:
            failCommands: ["commitTransaction"]
            closeConnection: true
```
To:
```
      - name: targetedFailPoint
        object: testRunner
        arguments:
          session: session0
          failPoint:
            configureFailPoint: failCommand
            mode: { times: 1 }
            data:
              failCommands: ["commitTransaction"]
              closeConnection: true
```

I prefer the newer format so I'm going to keep it unless there are objections.

Reply:
I think borrowing existing `object:` syntax is sensible, so I'm :+1: on keeping this.

I don't follow where "targeted" comes from, unless that refers to "targeting" the pinned mongos for the session argument. I don't think it would be targetted outside of a mongos topology, though. Do you plan that operation name being used exclusively for mongos topologies?

I imagine down the line there may be cases where we'd want to target a specific replica set member -- could "targetedFailPoint" be re-used there as well? If so, I might suggest replacing the `session` argument here with `target` and still provide a session object. For a mongos topology, you could document that the target is inferred from the pinned mongos within the session, and we could define a different interpretation as needed down the line. LGTM if you prefer this as-is, though. If kept as-is, perhaps we could use something like `readPreference` to handle replica set targeting down the line (if the need arises).

Reply:
> I don't follow where "targeted" comes from, unless that refers to "targeting" the pinned mongos for the session argument. 

The failpoint is targeted to the pinned session. I can rename to `mongosFailPoint` or `pinnedFailPoint` if you prefer, I just want to distinguish it from the higher level `failPoint` directive.

> Do you plan that operation name being used exclusively for mongos topologies?

I don't foresee using this outside of sharded transactions in the 4.2 timeframe so I don't think we should put too much time into thinking about how it should work on other topologies. A new spec can introduce a new test operation if/when they need it.

Reply:
I'm fine with keeping this as-is. Thanks for explaining the design rationale!